### ISSUE TITLE
Bug&Issue review - ChaletOS (Ubuntu 14.04) 64 bit and Deepin 15 (Debian unstable)

### ISSUE THREAD

**Issue by shadowvzs** [Return-To-The-Roots/s25client#404]
game version: 20160307
favorite test map: Wasserwege (sea map)

- in unlimited game the compute name too long (ex. "Computer (AI) easy" longer then button width)
- sound not work 
(in terminal)

> searching for drivers in ./share/s25rttr/driver/audio/*.so
> 0 audio drivers found!

but libaudioSDL.so is there, also i changed permission to enable read/write still same error 

- in game the "load game" button in options not work (click on it but no effect), sometimes make it freeze for few second after click to "load game" and game shut down without error message
(idk where can't find the crash log)

- in game need press multiple time "v" for speed up the game, so not allways have effect when pressing "v" on keyboard

- not bug but could be nice if "sea fight" disabled at addons then disable at harbour the "explorer expedition" or make a message about need enable it for dummy players like me, or make it default enabled "harbor block" for those like me, coz wondered after installed the game why can't do nothing on island maps with default settings

- not sure bug but sometimes takes ~10-15min for able make sea expedition (tryed launch expedition from 5-6 harbour same time, neither worked) and was free ships, maybe no explorer units but sometimes was 2 explorer ship from 10 harbour
have something issue with recruiting explorers?

- i dunno if normal but for me look like with non exhaustable mines my economy became slow, built ~14 farm or not 20 still got preoblem with makeing soldiers (3 armoury/2 brew etc), i don't know if normal, i curious if other people have similiar issue

- could be nice improvement if have a longer "sea explorer expedition" because if many island sometimes need hours for able reach the desired island

other side for me look like work well on linux, nice speed aswell

**Comment by Flamefire** [comment]
Thanks for the review.
Please open one issue per problem so we can individually address them as often some discussion is required and it will get very confusing with multiple things in one issue.

>  in unlimited game the compute name too long (ex. "Computer (AI) easy" longer then button width)
    sound not work (in terminal)

This is known. No fix by now but you can just use a bigger window

>    searching for drivers in ./share/s25rttr/driver/audio/*.so
    0 audio drivers found!

Try a `ldd libaudioSDL.so` to see if you are missing some dependencies

>    in game the "load game" button in options not work (click on it but no effect), sometimes make it freeze for few second after click to "load game" and game shut down without error message
    (idk where can't find the crash log)

This is not implemented yet. The freeze/crash is something else. The error log is in ~/.rttr/LOGS

>    in game need press multiple time "v" for speed up the game, so not allways have effect when pressing "v" on keyboard

It switches the speed one by one but you have to wait, till the switch reached the server. Not sure if this will be changed

>    not bug but could be nice if "sea fight" disabled at addons then disable at harbour the "explorer expedition" or make a message about need enable it for dummy players like me, or make it default enabled "harbor block" for those like me, coz wondered after installed the game why can't do nothing on island maps with default settings

Probably harbors should be disabled if sea fight is disabled. Would that work?

>    not sure bug but sometimes takes ~10-15min for able make sea expedition (tryed launch expedition from 5-6 harbour same time, neither worked) and was free ships, maybe no explorer units but sometimes was 2 explorer ship from 10 harbour
    have something issue with recruiting explorers?

Please submit a replay (also in ~/.rttr/REPLAYS) where this behaviour is visible.

>    i dunno if normal but for me look like with non exhaustable mines my economy became slow, built ~14 farm or not 20 still got preoblem with makeing soldiers (3 armoury/2 brew etc), i don't know if normal, i curious if other people have similiar issue

Please also submit a replay and savegame where one can see this issue

>    could be nice improvement if have a longer "sea explorer expedition" because if many island sometimes need hours for able reach the desired island

Not sure if this would not destroy the game experience... You can however increase the ship speed with the addon   
@Spikeone What do you think?

**Comment by shadowvzs** [comment]
can;t stay on this?

today i wanted retry the load game crash but now can't even access the previous saves, if i click to load  from main menu then totatlly empty and resume game say "The specified file can't be loaded"

log:
> lade "./share/s25rttr/RTTR/MUSIC/S2_Standard.pll"
> 3 in_order
> SavedFile::Load: ERROR: Old file version (version: 30; expected: 31)!
> SavedFile::Load: ERROR: Old file version (version: 30; expected: 31)!
> SavedFile::Load: ERROR: Old file version (version: 30; expected: 31)!
> SavedFile::Load: ERROR: Old file version (version: 30; expected: 31)!
> SavedFile::Load: ERROR: Old file version (version: 30; expected: 31)!
> SavedFile::Load: ERROR: Old file version (version: 30; expected: 31)!
> SavedFile::Load: ERROR: Old file version (version: 30; expected: 31)!
> SavedFile::Load: ERROR: Old file version (version: 30; expected: 31)!
> SavedFile::Load: ERROR: Old file version (version: 30; expected: 31)!
> SavedFile::Load: ERROR: Old file version (version: 30; expected: 31)!
> SavedFile::Load: ERROR: Old file version (version: 30; expected: 31)!
> SavedFile::Load: ERROR: Old file version (version: 30; expected: 31)!
> SavedFile::Load: ERROR: Old file version (version: 30; expected: 31)!
> SavedFile::Load: ERROR: Old file version (version: 30; expected: 31)!
> SavedFile::Load: ERROR: Old file version (version: 30; expected: 31)!
> SavedFile::Load: ERROR: Old file version (version: 30; expected: 31)!
> SavedFile::Load: ERROR: Old file version (version: 30; expected: 31)!
> SavedFile::Load: ERROR: Old file version (version: 30; expected: 31)!
> SavedFile::Load: ERROR: Old file version (version: 30; expected: 31)!
> SavedFile::Load: ERROR: Old file version (version: 30; expected: 31)!

last save yesterday nightm still there 6 .sav file in directory (could be problem one have no name? simple ".sav" and not "xyz.sav"?)
so i made a new savegame and noticed the load button now removed in game menu, so i tryed in main menu and visible the new save game and work, still older are invisible

tryed ldd libaudioSDL.so command but now i am not smarter :)
>      shadowvzs@shadowvzs-desktop:/opt/s25rttr/share/s25rttr/driver/audio$ ldd libaudioSDL.so
> 	linux-vdso.so.1 =>  (0x00007ffec02f1000)
> 	libSDL-1.2.so.0 => /usr/lib/x86_64-linux-gnu/libSDL-1.2.so.0 (0x00007fb1e302f000)
> 	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fb1e2e11000)
> 	libSDL_mixer-1.2.so.0 => not found
> 	libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007fb1e2b0d000)
> 	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007fb1e28f7000)
> 	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fb1e2532000)
> 	libasound.so.2 => /usr/lib/x86_64-linux-gnu/libasound.so.2 (0x00007fb1e2242000)
> 	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fb1e1f3c000)
> 	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007fb1e1d38000)
> 	libpulse-simple.so.0 => /usr/lib/x86_64-linux-gnu/libpulse-simple.so.0 (0x00007fb1e1b34000)
> 	libpulse.so.0 => /usr/lib/x86_64-linux-gnu/libpulse.so.0 (0x00007fb1e18eb000)
> 	libX11.so.6 => /usr/lib/x86_64-linux-gnu/libX11.so.6 (0x00007fb1e15b6000)
> 	libXext.so.6 => /usr/lib/x86_64-linux-gnu/libXext.so.6 (0x00007fb1e13a4000)
> 	libcaca.so.0 => /usr/lib/x86_64-linux-gnu/libcaca.so.0 (0x00007fb1e10d7000)
> 	/lib64/ld-linux-x86-64.so.2 (0x00007fb1e34e1000)
> 	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007fb1e0ecf000)
> 	libpulsecommon-4.0.so => /usr/lib/x86_64-linux-gnu/pulseaudio/libpulsecommon-4.0.so (0x00007fb1e0c68000)
> 	libjson-c.so.2 => /lib/x86_64-linux-gnu/libjson-c.so.2 (0x00007fb1e0a5d000)
> 	libdbus-1.so.3 => /lib/x86_64-linux-gnu/libdbus-1.so.3 (0x00007fb1e0818000)
> 	libxcb.so.1 => /usr/lib/x86_64-linux-gnu/libxcb.so.1 (0x00007fb1e05f9000)
> 	libslang.so.2 => /lib/x86_64-linux-gnu/libslang.so.2 (0x00007fb1e0269000)
> 	libncursesw.so.5 => /lib/x86_64-linux-gnu/libncursesw.so.5 (0x00007fb1e0035000)
> 	libtinfo.so.5 => /lib/x86_64-linux-gnu/libtinfo.so.5 (0x00007fb1dfe0c000)
> 	libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007fb1dfbf3000)
> 	libwrap.so.0 => /lib/x86_64-linux-gnu/libwrap.so.0 (0x00007fb1df9e9000)
> 	libsndfile.so.1 => /usr/lib/x86_64-linux-gnu/libsndfile.so.1 (0x00007fb1df781000)
> 	libasyncns.so.0 => /usr/lib/x86_64-linux-gnu/libasyncns.so.0 (0x00007fb1df57b000)
> 	libXau.so.6 => /usr/lib/x86_64-linux-gnu/libXau.so.6 (0x00007fb1df377000)
> 	libXdmcp.so.6 => /usr/lib/x86_64-linux-gnu/libXdmcp.so.6 (0x00007fb1df171000)
> 	libnsl.so.1 => /lib/x86_64-linux-gnu/libnsl.so.1 (0x00007fb1def57000)
> 	libFLAC.so.8 => /usr/lib/x86_64-linux-gnu/libFLAC.so.8 (0x00007fb1ded26000)
> 	libvorbisenc.so.2 => /usr/lib/x86_64-linux-gnu/libvorbisenc.so.2 (0x00007fb1de857000)
> 	libvorbis.so.0 => /usr/lib/x86_64-linux-gnu/libvorbis.so.0 (0x00007fb1de62a000)
> 	libogg.so.0 => /usr/lib/x86_64-linux-gnu/libogg.so.0 (0x00007fb1de421000)
> 	libresolv.so.2 => /lib/x86_64-linux-gnu/libresolv.so.2 (0x00007fb1de206000)


**This is known. No fix by now but you can just use a bigger window**
can't simple rename to "Comp" (ex. Comp easy) like in game where we can compare the kingdoms teritory/military etc with eachother? i think simple and best way

**Probably harbors should be disabled if sea fight is disabled. Would that work?**
same result, sea maps unable with with sea fight, if u disable harbour then same you can't win map and still player don't know why he can't build harbour also in original campaign map have map where was usefull the harbour without sea fight, in custom sea maps where annoying issue, can't insert tooltip or something in game message what warm player about it disable or just enable by default? this solve the real problem

**It switches the speed one by one but you have to wait, till the switch reached the server. Not sure if this will be changed**
i waited seconds still sometimes not react to pressing "v"

**Please submit a replay (also in ~/.rttr/REPLAYS) where this behaviour is visible.**
2h343 submit save and replay, saves are more than few mb

**Comment by Flamefire** [comment]
The savegame version was changed with the latest build, this is what the console tells you. You could still download the old version from the download page and play with that if you need the old saves. So this is correct behaviour.

Look at the output: `libSDL_mixer-1.2.so.0 => not found` Just install libSDL_mixer and you are done

The button issue is already fixed now.

@Spikeone @MarcusSt @Flow86 What about enabling seafaring by default? If there are no harbors on the map it should not matter and I understand the confusion if you can build harbors but not use them.

Speed change: Check the console. It tells you when the speed was actually changed... Still this should be improved...

> 2h343 submit save and replay, saves are more than few mb
Excuse me?
What you need to do if you have a bug where the game is doing something wrong:
Open the replay and play it to check if you see the same thing there. If yes, upload it somewhere and post a link here. If possible, also include a savegame where this is shown and tell what to do there (e.g. after loading build xyz)

**Comment by shadowvzs** [comment]
speed up game stuff, now i checked terminal, each time appear something in terminal when i press "v" just in game not appear the speedUp icon in right-top corner, please check the video, this is how must it work?
http://1drv.ms/1M4yR4U

> @Spikeone @MarcusSt @Flow86 What about enabling seafaring by default? If there are no harbors on the map it should not matter and I understand the confusion if you can build harbors but not use them.

i agree with enableing it defaultly the sea fight solve every problem, honestly, thats why i liked better normal settler2 vs gold version because s2 was with harbours and thats what very lovely in this game.


so "apt-get install libSDL_mixer" not work (E: Couldn't find package libSDL_mixer) but "sudo apt-get install libsdl-mixer1.2" worked

> Reading package lists... Done
> Building dependency tree       
> Reading state information... Done
> The following packages were automatically installed and are no longer required:
>   fonts-horai-umefont fonts-wqy-microhei gnome-exe-thumbnailer libgif4:i386
>   libgstreamer-plugins-base0.10-0:i386 libgstreamer0.10-0:i386
>   linux-image-3.13.0-61-generic linux-image-extra-3.13.0-61-generic
>   ocl-icd-libopencl1 ocl-icd-libopencl1:i386 p11-kit-modules:i386 p7zip
>   ttf-wqy-microhei unixodbc wine-gecko2.34 wine-gecko2.34:i386 wine-mono4.5.4
>   winetricks
> Use 'apt-get autoremove' to remove them.
> The following NEW packages will be installed:
>   libsdl-mixer1.2
> 0 upgraded, 1 newly installed, 0 to remove and 131 not upgraded.
> Need to get 72,1 kB of archives.
> After this operation, 245 kB of additional disk space will be used.
> Get:1 http://us.archive.ubuntu.com/ubuntu/ trusty/universe libsdl-mixer1.2 amd64 1.2.12-10 [72,1 kB]
> Fetched 72,1 kB in 1s (61,5 kB/s)                      
> Selecting previously unselected package libsdl-mixer1.2:amd64.
> (Reading database ... 369029 files and directories currently installed.)
> Preparing to unpack .../libsdl-mixer1.2_1.2.12-10_amd64.deb ...
> Unpacking libsdl-mixer1.2:amd64 (1.2.12-10) ...
> Setting up libsdl-mixer1.2:amd64 (1.2.12-10) ...
> Processing triggers for libc-bin (2.19-0ubuntu6.6) ...

still no sound after launch it, following error (btw could u make a .sh script for linux users what check if its installed and if not then install it?)

> 1 audio drivers found!
> Failed to access the SoundFont /usr/share/sounds/sf2/FluidR3_GM.sf2
> Writing "/home/shadowvzs/.s25rttr/CONFIG.INI": finished

probabil because at usr/share/sounds/ no sf2 directory
screenshot from folder and game replay is here 
http://1drv.ms/1M4yR4U

so i googled and used this in terminal "sudo apt-get install fluid-soundfont-gm fluid-soundfont-gs"
this downloaded around 121mb,

> Reading package lists... Done
> Building dependency tree       
> Reading state information... Done
> The following packages were automatically installed and are no longer required:
>   fonts-horai-umefont fonts-wqy-microhei gnome-exe-thumbnailer libgif4:i386
>   libgstreamer-plugins-base0.10-0:i386 libgstreamer0.10-0:i386
>   linux-image-3.13.0-61-generic linux-image-extra-3.13.0-61-generic
>   ocl-icd-libopencl1 ocl-icd-libopencl1:i386 p11-kit-modules:i386 p7zip
>   ttf-wqy-microhei unixodbc wine-gecko2.34 wine-gecko2.34:i386 wine-mono4.5.4
>   winetricks
> Use 'apt-get autoremove' to remove them.
> Suggested packages:
>   fluidsynth timidity
> The following NEW packages will be installed:
>   fluid-soundfont-gm fluid-soundfont-gs
> 0 upgraded, 2 newly installed, 0 to remove and 131 not upgraded.
> Need to get 121 MB of archives.
> After this operation, 152 MB of additional disk space will be used.
> Get:1 http://us.archive.ubuntu.com/ubuntu/ trusty/universe fluid-soundfont-gm all 3.1-5 [119 MB]
> Get:2 http://us.archive.ubuntu.com/ubuntu/ trusty/universe fluid-soundfont-gs all 3.1-5 [2.560 kB]
> Fetched 121 MB in 1min 41s (1.197 kB/s)                                        
> Selecting previously unselected package fluid-soundfont-gm.
> (Reading database ... 369036 files and directories currently installed.)
> Preparing to unpack .../fluid-soundfont-gm_3.1-5_all.deb ...
> Unpacking fluid-soundfont-gm (3.1-5) ...
> Selecting previously unselected package fluid-soundfont-gs.
> Preparing to unpack .../fluid-soundfont-gs_3.1-5_all.deb ...
> Unpacking fluid-soundfont-gs (3.1-5) ...
> Setting up fluid-soundfont-gm (3.1-5) ...
> Setting up fluid-soundfont-gs (3.1-5) ...

then i got this error at s25rttr

> searching for drivers in ./share/s25rttr/driver/audio
> 1 audio drivers found!
> Failed to access the SoundFont /usr/share/sounds/sf2/TimGM6mb.sf2

so i installed this
sudo apt-get install musescore-soundfont-gm

this around 5-6mb package, now music work

so basically i installed this 3 package
sudo apt-get install libsdl-mixer1.2
sudo apt-get install fluid-soundfont-gm fluid-soundfont-gs
sudo apt-get install musescore-soundfont-gm

idk if usefull but i uploaded the whole sf2 directory
usr.7z - http://1drv.ms/1M4yR4U

**Comment by shadowvzs** [comment]
So i decided test on my other installed linux, Deepin 15 what based directly to debian unstable, so i copied these folder what mentioned above (sf2 what archived in usr.7z)

tryed and same driver problem than at chaletos then i used:
> sudo apt-get install libsdl-mixer1.2

then worked flawless :) 
(btw i simple copied whole s25rttr directory from ChaletOS to Deepin, worked nicely)

video is here:
http://1drv.ms/1M4yR4U

**Comment by Flamefire** [comment]
As you see it is already getting confusing in here and working on one issue is  pretty hard now.

About the speed thing: The running-icon in the corner appears only on very fast speed. We have more settings than the original did. This is why it seems, like nothing is happening. But check the GF-length on the top

**Comment by Flamefire** [comment]
I opened an issue for the speed: #406 
If you have more information on the other issues, please open one issue for each of them. As some bigger things from this are resolved, I'll close this.
